### ISSUE TITLE
Choosing group members change the search input and it's label style

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22812]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1.Click on the FAB and create New group
2.Start adding the group members and notice the search input and it's label

## Expected Result: 
Choosing group members shouldn't change the search input and it's label style for better UX.
## Actual Result: 
Every time we choose a member we change the search input and it's label, because of the search input refocus

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.40-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[Screencast from 13-07-23 05_23_15.webm](https://github.com/Expensify/App/assets/43996225/6a9b1ffd-30fb-4213-bf33-7b85c583d279)

https://github.com/Expensify/App/assets/43996225/358c55d1-b9d8-445f-8241-26c76b07f216

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Ahmed-Abdella
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689216766939829

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Toggling group members changes the search input and its label style

### What is the root cause of that problem?

When user selects the members, search input lose the focus, and the focus is set here again
https://github.com/Expensify/App/blob/28acc0267738c3c9305c412cd56d5f93ab8066b3/src/components/OptionsSelector/BaseOptionsSelector.js#L267

The search input changes if it loses the focus. Refocusing is the root cause

### What changes do you think we should make in order to solve the problem?

We can prevent the lost of focus by preventing default action of mouse events
Add the following code above [this line](https://github.com/Expensify/App/blob/28acc0267738c3c9305c412cd56d5f93ab8066b3/src/components/OptionsList/BaseOptionsList.js#L239)

```
            onMouseDown={(e) => e.preventDefault()}
            onMouseUp={(e) => e.preventDefault()}
```

This works as expected.

<details>
    <summary>Result</summary>


</details>

### What alternative solutions did you explore? (Optional)

**Comment by Ahmed-Abdella** [comment]
Also happens in split bill
[Screencast from 15-07-23 05:02:59.webm](https://github.com/Expensify/App/assets/77763699/588c5c71-e9fb-4925-b896-8365149be398)

**Comment by michaelhaxhiu** [comment]
Reproduced easily. And it has the same effect on desktop, so I'm gonna check that box in the **Platforms:** section of the GH.

**Comment by michaelhaxhiu** [comment]
Lol why does it say I added the `Overdue` label back... Melvin you lie.

**Comment by Talha345** [comment]
> ### What is the root cause of that problem?

Your RCA is correct but only `onMouseDown` will suffice.

**Comment by s-alves10** [comment]
@Talha345 
Maybe. But it might cause other issues as in https://github.com/Expensify/App/issues/21727

**Comment by michaelhaxhiu** [comment]
@0xmiroslav can you please review the proposal provided?

**Comment by michaelhaxhiu** [comment]
If none are applicable, I will double the price of this job.

**Comment by s-alves10** [comment]
> still looking for ideal solution

We're using this way in [BaseTextInput](https://github.com/Expensify/App/blob/main/src/components/TextInput/BaseTextInput.js) and [ReportActionCompose](https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionCompose.js) and other several places. I think this is a good choice

**Comment by michaelhaxhiu** [comment]
@0xmiroslav any update on the latest proposals?

**Comment by michaelhaxhiu** [comment]
Note: I'm preparing to go OOO for ~2 weeks, and need a BZ buddy to oversee this while I'm gone.

The next step is for us to field proposals until a winning one is chosen by the C+ @0xmiroslav. If no winning proposals by 08/07, you can double price again from $2k -> $4k to get more activity here.

Thanks in advance @tjferriss

**Comment by Talha345** [comment]
@michaelhaxhiu the C+ member has been MIA and  hasn't provided any feedback on the previous proposals!

**Comment by Talha345** [comment]
> @Talha345 your proposal changes the behavior. I don't think we should lose focus on search while choosing group members if already focused.

Well, it is the default behavior and as I mentioned in my proposal that we cannot predict the user's next action so IMHO it doesn't make sense to refocus it after every member selection.

**Comment by s-alves10** [comment]
@0xmiroslav 

My solution doesn't refocus(what @Talha345 is concerned), rather prevent losing focus. What's the drawback of my solution?

**Comment by Talha345** [comment]
If we want to prevent loss of focus in the first place, @s-alves10 's first solution makes the most sense to me.We should keep it simple in this case and not create an entire state just for this purpose and bloat the code unnecessarily.

**Comment by tjferriss** [comment]
@0xmiroslav can you take a look at the above comments and offer some guidance? Thanks!

**Comment by situchan** [comment]
I'd like to take this as C+ based on [discussion](https://expensify.slack.com/archives/C02NK2DQWUX/p1691641008846119?thread_ts=1691488353.531539&cid=C02NK2DQWUX)

**Comment by s-alves10** [comment]
@situchan 

Please let me know your though on [my proposal](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613)

**Comment by aditygrg2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Avoiding label style and search input on selecting group numbers on the `BaseOptionsSelector` component.

### What is the root cause of that problem?

https://github.com/Expensify/App/blob/a03a21d25832760cf9db4235b0fbc88cc08bd490/src/components/OptionsSelector/BaseOptionsSelector.js#L333C1-L338C19

The `onBlur` event returns here if `this.props.shouldFocusOnSelectRow` is false. This is incorrect because if any `selectRow` is focused then `onBlur` event should not switch the `currentTarget`.
This will make sure that it will not break on any component using `BaseOptionsSelector`

### What changes do you think we should make in order to solve the problem?

If this can be toggled, i.e., returning when this is true and only changing focus when this is false, will solve the problem.

### What alternative solutions did you explore? (Optional)
NA

Results:

https://github.com/Expensify/App/assets/98523623/03c329f7-765a-444c-b107-0d24c1d149fc

**Comment by situchan** [comment]
@trjExpensify can you please assign me here?
And I suggest to put this on hold for #22856 as the select behavior might change.

**Comment by trjExpensify** [comment]
I think that's for @tjferriss :)

**Comment by situchan** [comment]
> I think that's for @tjferriss :)

Ah sorry meant to tag @tjferriss :)

**Comment by tjferriss** [comment]
@situchan you're assigned.

**Comment by situchan** [comment]
The holding [issue](https://github.com/Expensify/App/issues/22856) will be closed as won't fix.
We can now resume this issue.

**Comment by Talha345** [comment]
@situchan Can you please review the proposals above and let us know your thoughts?

**Comment by tjferriss** [comment]
@situchan friendly reminder on the proposals.

@michaelhaxhiu do you want to take this back over now that you're back from OOO?

**Comment by situchan** [comment]
@tjferriss let's remove Hold from title

**Comment by situchan** [comment]
@s-alves10's [proposal](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613) looks good to me.
I don't see any other better solution to prevent blur on focused search input.

🎀 👀 🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
Reviewing now

**Comment by MonilBhavsar** [comment]
Proposal looks good, but I feel it is not necessary or required to keep focus on search text input, while options are being selected.

looks like we have that in Workspace > Invite members. We don't keep focus on input while selecting workspace invitees. We can do similar here, I think cc @tjferriss

**Comment by situchan** [comment]
@MonilBhavsar Invite page (uses SelectionList) is different than search/new chat/new group pages (uses OptionsList).
There was full discussion search and option focus behavior here - https://expensify.slack.com/archives/C01GTK53T8Q/p1692122623311399?thread_ts=1689954073.865489&cid=C01GTK53T8Q.
And also updated test rail - https://github.com/Expensify/Expensify/issues/311312

**Comment by Talha345** [comment]
> Proposal looks good, but I feel it is not necessary or required to keep focus on search text input, while options are being selected.

@MonilBhavsar Yes that is what I suggested too because it feels kinda hacky to me to persist the focus even though the user has moved to another element and we do not know if the user will come back again to the search field or not.I have explained this in detail in my proposal https://github.com/Expensify/App/issues/22812#issuecomment-1658437718.

@situchan If it is decided to move forward with what @MonilBhavsar and I suggested, you can have a look at my proposal using which we can achieve this with minimal changes.

**Comment by michaelhaxhiu** [comment]
Taking this back over. @MonilBhavsar do you agree with @situchan's comment (above) about how the SelectionList =/= OptionsList, and thus the behavior may not need to be equivalent. The next step lies with you to unblock us :)

**Comment by MonilBhavsar** [comment]
Commented in the thread. Looks like all agree with keeping dual focus. I think then we should do it similar for all elements across the App https://expensify.slack.com/archives/C01GTK53T8Q/p1692122623311399?thread_ts=1689954073.865489&cid=C01GTK53T8Q

**Comment by Talha345** [comment]
> Commented in the thread. Looks like all agree with keeping dual focus. I think then we should do it similar for all elements across the App https://expensify.slack.com/archives/C01GTK53T8Q/p1692122623311399?thread_ts=1689954073.865489&cid=C01GTK53T8Q

@MonilBhavsar Apart from this,everywhere we have the standard behavior.I think we should re-think this and instead of changing the behavior at all other places, we should make the behavior in this issue consistent with the entire app and not vice versa.Just my 2 cents.

**Comment by situchan** [comment]
@Talha345 There are only 2 components which have both search input and selectable list: OptionsList, SelectionList.
They represent the entire app.

**Comment by MonilBhavsar** [comment]
Agree, @Talha345 feel free to share your thoughts in the slack thread

**Comment by Talha345** [comment]
> @Talha345 There are only 2 components which have both search input and selectable list: OptionsList, SelectionList.
> They represent the entire app.

@situchan Yes you are right! I was simply talking in terms of behavior without going into the technical terminology to make it consistent across the app.Having focus on multiple elements at once doesn't make sense to me logically but we can proceed with whatever the team decides.

**Comment by s-alves10** [comment]
I think keeping focus in search input is a good choice from the view of UX

**Comment by MonilBhavsar** [comment]
Right, but looks like we want to do something new from industry standard, and allow users to keep typing while also navigate through the list options.

**Comment by kameshwarnayak** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
Choosing group members change the search input and it's label style

### What is the root cause of that problem?
The root cause of the issue is that when the user clicks on the list item, the focus moves out of the `TextInput` which is the default behaviour. We focus back on the `TextInput` in the line below. The time-lag between blur and focus causes the flicker. 

https://github.com/Expensify/App/blob/57b76067cd3d7b59e84c4a387d11ad33e6a97d05/src/components/OptionsSelector/BaseOptionsSelector.js#L302 

### What changes do you think we should make in order to solve the problem?

1. `TextInput` component already has `forceActiveLabel` prop that allows us to keep the label active. We can use that to prevent the label flickering. 
2. We can define a new state variable `keepActive` in `BaseOptionsSelector.js` and pass it to `TextInput` component. 
3. Set `keepActive` to true when `TextInput` is focused
4. `keepActive` is set to false when users click outside the `TextInput` or the Checklist
5. In `BaseTextInput.js` apply `styles.borderColorFocus` if `keepActive` is true

_Idea is to use the existing options in the application to achieve the result with minimal change._  

The following code changes can be made to achieve this.

[BaseOptionsSelector.js](https://github.com/Expensify/App/blob/57b76067cd3d7b59e84c4a387d11ad33e6a97d05/src/components/OptionsSelector/BaseOptionsSelector.js)

```
     /**
     * @param {Boolean} status
     */
    updateKeepActive(status) {
        this.setState({keepActive: status});
    }
    ...
    <TextInput
        keepActive={this.state.keepActive}
        forceActiveLabel={this.state.keepActive}
        onFocus={() => this.updateKeepActive(true)}
        ...
    />
    ...
    <View
        style={[styles.flexGrow1, styles.flexShrink1, styles.flexBasisAuto]}
        onClick={() =>this.updateKeepActive(false)}
    >

```
[BaseTextInput.js ](https://github.com/Expensify/App/blob/57b76067cd3d7b59e84c4a387d11ad33e6a97d05/src/components/TextInput/BaseTextInput.js#L235)
```
    const textInputContainerStyles = StyleSheet.flatten([
        ...
        (props.keepActive || (!props.hideFocusedState && isFocused)) && styles.borderColorFocus,
        ...
    ]);

```


Result :


https://github.com/Expensify/App/assets/2232432/62ea888e-582e-419d-8e36-17cc37d93092



### What alternative solutions did you explore? (Optional)
None

**Comment by situchan** [comment]
@kameshwarnayak thanks for your proposal. Label seems to be in fixed position but input cursor still flickers.
I don't think your solution is better and straightforward than preventDefault approach.

**Comment by MonilBhavsar** [comment]
Still waiting for a correct proposal to generalize things as we discussed in slack.

**Comment by situchan** [comment]
> Still waiting for a correct proposal to generalize things as we discussed in slack.

@MonilBhavsar can you please share the expected behavior finally landed on?

**Comment by s-alves10** [comment]
> Still waiting for a correct proposal to generalize things as we discussed in slack.

What is the expected behavior? If something was changed, please share here

**Comment by MonilBhavsar** [comment]
It is in the thread that was linked here https://github.com/Expensify/App/issues/22812#issuecomment-1702566251

We should fix it in more wider way and not just for this group members list. Also, we should explore having a generic component except two different components - OptionsList and SelectionsList

Please see this comment from tgolen in the thread. I agree with this
> I'd love for both pages to have consistent UX and I think we should take the time right now to make them such. Is the proposal to make a generic component that can be used on both pages?

And we had couple of thumbs up on this -

> Looks like, we agree with doing something different and focussing on both inputs at the same time, then I agree with the point Tim brought above to use similar component or style and have consistency across all elements in the App. If we agree having focus on two elements, then I think we should do it now for all usages inside the App.

**Comment by MonilBhavsar** [comment]
Waiting for proposal here

**Comment by jscardona12** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Choosing group members change the search input and it's label style
### What is the root cause of that problem?
#### Note
The UI is diffrent now than when the error was reported.
Now we have same menu for startign a chat or a group.

Also now in Main the search bar is not being refocus.
<details>
<summary>How its working</summary>
https://github.com/Expensify/App/assets/16781411/2be83dc2-8430-4345-a86b-268ad84e913f
</details>

When user select members click on add to group or unselect the selected member focus is lost on search bar.
### What changes do you think we should make in order to solve the problem?
In this lines setUp  `onMouseDown={(e)=>{e.preventDefault()}}` to prevent onfocus on the search bar.

https://github.com/Expensify/App/blob/9533031c7795567392cca0b295212b342c9acdfe/src/components/OptionRow.js#L279-L284
and in 
https://github.com/Expensify/App/blob/9533031c7795567392cca0b295212b342c9acdfe/src/components/OptionRow.js#L286-L291

<details>
<summary>Solution Working</summary>
https://github.com/Expensify/App/assets/16781411/6243b205-eb37-4377-81f8-4848aa2e7670
</details>

### Refactor to use OptionsList instead of SelectionsList
Right now the Selections List is being used in the following components
 - https://github.com/Expensify/App/blob/main/src/components/NewDatePicker/CalendarPicker/YearPickerModal.js
 - https://github.com/Expensify/App/blob/main/src/components/CountryPicker/CountrySelectorModal.js
 - https://github.com/Expensify/App/blob/main/src/components/StatePicker/StateSelectorModal.js
 - https://github.com/Expensify/App/blob/main/src/pages/settings/Preferences/LanguagePage.js
 - https://github.com/Expensify/App/blob/main/src/pages/settings/Preferences/PriorityModePage.js
 - https://github.com/Expensify/App/blob/main/src/pages/settings/Preferences/ThemePage.js
 - https://github.com/Expensify/App/blob/main/src/pages/settings/Profile/PronounsPages.js
 - https://github.com/Expensify/App/blob/main/src/pages/settings/Profile/TimezoneSelectPages.js
 - https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceInvitePage.js
 - https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceMemebersPage.js

We need to change the SelectionsList for the OptionsList in this components. It will require de development of the select All in the Options List. and to test thorughly the features.

@MonilBhavsar currently the UX of both components are diffrent are we goign with the UX of OptionsList correct? 

### What alternative solutions did you explore? (Optional)

**Comment by MonilBhavsar** [comment]
We recently changed the workflow around creating groups and this issue is not visible any more.
The options list in Creating group and splitting money and selection in workspace member settings loses focus when we try to select the options.

If we want to pursue the idea of keeping the focus on search while selecting the option, that could be a feature request I think. @michaelhaxhiu should we consider working on this issue as a feature request, or close it and open new issue, as this bug is not reproducible anymore.

**Comment by situchan** [comment]
The pages have been refactored but we can still apply solution to these badges:

https://github.com/Expensify/App/assets/108292595/950757e9-52f0-4b08-85cf-26c932a74263

**Comment by situchan** [comment]
And that's what native app already does:


https://github.com/Expensify/App/assets/108292595/f3a6c2d1-6b33-4f8f-9ec1-a5932daa722c



So good to fix for consistency

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613)

**Comment by MonilBhavsar** [comment]
@situchan  Sorry, I didn't get 
The issue reported originally is not reproducible, is it?


https://github.com/Expensify/App/assets/32012005/5cf27831-44c2-4ef5-a316-7d39e83c5cad

**Comment by situchan** [comment]
> The issue reported originally is not reproducible, is it?

Correct because UX changed and selecting row navigates to another page always.

But code still remaining

https://github.com/Expensify/App/blob/41ddd7428340fd8d347e8160980809e76a93899b/src/pages/NewChatPage.js#L201

**Comment by jscardona12** [comment]
@MonilBhavsar it is reproducible, as i show on the videos of my [Proposal] (https://github.com/Expensify/App/issues/22812#issuecomment-1727743878)

**Comment by michaelhaxhiu** [comment]
I'm confused myself. Does the bug still exist and need fixing? Or are we saying the bug is fixed from UX perspective and we just need to clean up the code now? 

cc @MonilBhavsar

**Comment by situchan** [comment]
This is not reproducible because of recent regression.
Context: https://expensify.slack.com/archives/C01GTK53T8Q/p1695863206217249?thread_ts=1689954073.865489&cid=C01GTK53T8Q
Seems like we're going to fix as bug

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this to another BZ as part of my preparation for Sabbatical (starting Friday).

**Next steps:**

- Sherpa this forward with @situchan and @MonilBhavsar
- After that [regression](https://github.com/Expensify/App/issues/22812#issuecomment-1739197614) is fixed, this bug will return to exist and so we may need to wait a few days till that it shows itself again.

**Comment by MonilBhavsar** [comment]
Looks like the regression is because of https://github.com/Expensify/App/pull/25564/files

**Comment by sonialiap** [comment]
@MonilBhavsar am I reading this correctly that the regression has not been corrected? If it hasn't, are we solving it in this issue or do we need to locate/open a separate issue for the regression before fixing this bug?

**Comment by situchan** [comment]
There's no GH created and no one is working on it to fix that "regression". I think we can fix here.

**Comment by Ahmed-Abdella** [comment]
@sonialiap @situchan This bug can now be reproduced again. I believe the regression responsible for its disappearance has been fixed.

[Screencast from 04-10-23 08:43:39.webm](https://github.com/Expensify/App/assets/77763699/ac3adfe0-24a1-47be-af93-86ef36844784)

**Comment by situchan** [comment]
@Ahmed-Abdella Ah, good catch!
I just confirmed that [PR](https://github.com/Expensify/App/pull/28423) which fixes https://github.com/Expensify/App/issues/28070 was merged 2 days ago, which made this bug reproducible again.

**Comment by situchan** [comment]
@s-alves10 can you update your proposal based on latest codebase?

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613)

**Comment by situchan** [comment]
@s-alves10's [updated proposal](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613) looks good.
cc: @MonilBhavsar 

🎀 👀 🎀 C+ reviewed

**Comment by MonilBhavsar** [comment]
@s-alves10 @situchan this does not persist focus on Invite members page that uses selectionList, right?

**Comment by situchan** [comment]
> @s-alves10 @situchan this does not persist focus on Invite members page that uses selectionList, right?

right. We can apply same logic if needed, but that page always not persisted focus from the beginning.

**Comment by situchan** [comment]
Oh I just noticed that refocus logic implemented in https://github.com/Expensify/App/pull/27497 and merged recently.
So now this bug happens on selection list as well.
@s-alves10 you may need to update your proposal again to cover these pages which use selection list.

**Comment by situchan** [comment]
So good timing of this issue to start

**Comment by s-alves10** [comment]
Okay. I'll update soon

**Comment by MonilBhavsar** [comment]
> So good timing of this issue to start

Right 👍 

> Okay. I'll update soon

Thank you! Looking forward to it. Would be great to avoid code deduplication i.e. not pass similar props and handle the logic in two different components in a similar way.

**Comment by s-alves10** [comment]
@situchan @MonilBhavsar 

## Proposal
[Updated](https://github.com/Expensify/App/issues/22812#issuecomment-1634440613)

**Comment by situchan** [comment]
> Would be great to avoid code deduplication i.e. not pass similar props and handle the logic in two different components in a similar way.

@s-alves10 does your proposal take this feedback into account?

**Comment by s-alves10** [comment]
Both `SelectionList` and `OptionsSelector` have `shouldFocusOnSelectRow` props. I used them.
I don't think deduplication is needed

**Comment by situchan** [comment]
Currently, there's no refocus logic on Invite new members page.
https://github.com/Expensify/App/pull/27497 handled Members page only.
I think we can fix it here.

@MonilBhavsar I am fine with updated proposal.

**Comment by MonilBhavsar** [comment]
Thank you @s-alves10 and @situchan for carrying this forward!
Looks good to me 👍

**Comment by MonilBhavsar** [comment]
Why the offers were not sent automatically 
Because it was Internal, may be 🤔

**Comment by situchan** [comment]
> Why the offers were not sent automatically Because it was Internal, may be 🤔

@MonilBhavsar Now it's External & Help Wanted so unassign and reassign @s-alves10

**Comment by s-alves10** [comment]
@situchan 

PR is ready for review https://github.com/Expensify/App/pull/29084

**Comment by Ahmed-Abdella** [comment]
Sorry @MonilBhavsar , @sonialiap but this issue was reported before changing the rate, it should be at 250$.

**Comment by sonialiap** [comment]
@situchan review + bonus = $3000 - paid ✔️ 
@s-alves10 fix + bonus = $3000 - offer sent
@Ahmed-Abdella report = $250 - new offer sent with the correct amount

**Comment by Ahmed-Abdella** [comment]
@sonialiap I accepted the offer, Thanks.

**Comment by sonialiap** [comment]
I'm OOO Oct 16-23. Adding leave buddy

Status: waiting to issue payment https://github.com/Expensify/App/issues/22812#issuecomment-1765319825

**Comment by MonilBhavsar** [comment]
Bumping to weekly!

**Comment by bfitzexpensify** [comment]
Flicking this one back to daily so that it's in the right spot in my k2 dashboard

**Comment by bfitzexpensify** [comment]
Due 10/23

**Comment by MonilBhavsar** [comment]
Payment is due today. @situchan could you please take a look at the checklist

**Comment by MonilBhavsar** [comment]
False comment, Melvin

**Comment by sonialiap** [comment]
@situchan review + bonus = $3000 - paid ✔️ https://github.com/Expensify/App/issues/22812#issuecomment-1763272663
@s-alves10 fix + bonus = $3000 - paid ✔️  
@Ahmed-Abdella report = $250 - paid ✔️

**Comment by sonialiap** [comment]
@situchan last step is to do the [checklist](https://github.com/Expensify/App/issues/22812#issuecomment-1765319861). I scrolled through the issue and I don't think it's been filled out yet but let me know if I missed it

**Comment by MonilBhavsar** [comment]
@situchan could you please take a look at the checklist. Thanks!

**Comment by situchan** [comment]
There were several changes on this behavior but the original root cause existed from the beginning as it's web's default behavior to blur element when another one is clicked. So no regression.

This is minor UI issue so I don't think it's worth adding to regression test which should run on every deploy.
And the refactor will not likely to happen again so the fix won't be reverted.

**Comment by MonilBhavsar** [comment]
I would say we should add regression test, as we usually update callback functions with lists and could probably cause regression

**Comment by situchan** [comment]
### Regression Test Proposal

Case 1:
1. Go to FAB > Start chat
2. Focus on search input if not focused
3. Select multiple users by tapping "Add to group" button
4. Deselect users again by tapping ✅ button
5. Verify that the style of search input and label doesn't change

Case 2:
1. Go to FAB > Request money
2. Input any amount in manual tab and tap Next
3. Focus on search input if not focused
4. Select multiple users by tapping "Split" button
5. Deselect users again by tapping ✅ button
6. Verify that the style of search input and label doesn't change

**Comment by MonilBhavsar** [comment]
And also Workspace settings > Members page please

**Comment by situchan** [comment]
Updated.

**Comment by sonialiap** [comment]
New regression test addition submitted
